community_usecase/Profile_generation/run_profile_generation.py

```diff
@@ -0,0 +1,253 @@
+    print("List of queried URLs:")
+    logger.info("List of queried URLs:")
+    for url in queried_urls:
+        print(url)
+        logger.info(url)
+
+    last_html_content = None
+
+    for item in reversed(chat_history):
+        assistant_content = item.get("assistant", "")
+        if (
+            isinstance(assistant_content, str)
+            and "```html" in assistant_content
+        ):
+            matches = re.findall(
+                r"```html\s*(.*?)\s*```", assistant_content, re.DOTALL
+            )
+            if matches:
+                last_html_content = matches[-1]
+                break
+
+    if last_html_content:
+        with open(filename, "w", encoding="utf-8") as f:
+            f.write(last_html_content)
+        print(f"\033[92mExtracted HTML saved to {filename}\033[0m")
+    else:
+        print("\033[91mNo HTML block found in chat history.\033[0m")
+
+    with open("process_history.json", "w", encoding="utf-8") as f:
+        json.dump(chat_history, f, ensure_ascii=False, indent=2)
+
+    print("Records saved to process_history.json")
+    print("============ Analysis Complete ============\n")
+
+
+def run_profile_generation(task: str | None = None):
+    """Run the profile_generation with the given task.
+
+    Args:
+        task (str, optional): The personal information.
+        Defaults to an example task.
+    """
+    if not task:
+        task = """
+find Bernard-Ghanem's information and generate a html page based on these 
+websites:
+https://www.linkedin.com/in/bernardghanem/
+https://scholar.google.com/citations?hl=en&user=rVsGTeEAAAAJ&view_op=list_works
+https://x.com/bernardsghanem
+https://www.researchgate.net/profile/Bernard-Ghanem
+https://www.bernardghanem.com/curriculum-vitae
+https://www.bernardghanem.com/home
+        """
+    html_prompt = """
+Generate a professional and natural-looking personal profile HTML page based 
+on the provided information, following these detailed requirements:
+
+Use standard HTML5 structure.
+
+Layout should be clearly organized using <h1> for the main title, <h2> for 
+section headings, and <p> for paragraph text.
+
+Use proper <a href> hyperlinks for any external links (e.g., LinkedIn, 
+ResearchGate, Google Scholar).
+
+Write in a natural, flowing narrative style suitable for introducing the 
+person to a broad audience (no bullet points).
+
+Strictly follow the given content structure:
+
+Personal Information (Name, Positions, Contact( include social media), 
+Research Keywords)
+
+Summary (Short description of expertise and overall role)
+
+Biography (Detailed career history (can be found in Linkedin) and research 
+focus)
+
+Research Interests (Areas of active research)
+
+Awards and Distinctions (Honors and recognitions)
+
+Education (Academic degrees in order)
+
+Engage / Related Links (Professional profiles and external resources)
+
+The final HTML should look clean, modern, and ready to be used as a complete 
+professional personal page.
+    """
```

Comment:
Can we use `response_format` in `ChatAgent` to generate a pydantic object with required fields and use exact the same html template as https://cemse.kaust.edu.sa/profiles/ahmed-eltawil to render the information? Please do not use prompt to generate html.